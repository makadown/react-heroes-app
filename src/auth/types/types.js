/**
 * Contiene nombres de las acciones que podr√© "disparar"
 */
export const types = {
    login: '[Auth] Login',
    logout: '[Auth] Logout',
}