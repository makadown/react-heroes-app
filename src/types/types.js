export const Types = {
    login: '[auth] login',
    logout: '[auth] logout',
};