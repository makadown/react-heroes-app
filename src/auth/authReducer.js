import { Types } from '../types/types';

export const AuthReducer = (state = {}, action) => {
    switch (action.type) {
        case Types.login:
            return {
                ...action.payload,
                logged: true
            };
        case Types.logout:
            return {
                logged: false
            };

        default:
            return state;
    }
};