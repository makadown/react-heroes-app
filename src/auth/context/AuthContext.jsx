import React, { createContext } from 'react'

/**
 * Contexto de autenticaci√≥n usado por AuthProvider.jsx
 */
export const AuthContext = createContext();
