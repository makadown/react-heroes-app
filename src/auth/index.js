export { LoginPage } from './pages/LoginPage';
export {AuthContext} from './context/AuthContext';
export {AuthProvider} from './context/AuthProvider';
export {authReducer} from './context/authReducer';
export {types} from './types/types';