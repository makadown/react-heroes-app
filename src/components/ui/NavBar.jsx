import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">
        Asociaciones
      </Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>

      <div className="collapse navbar-collapse" id="navbarNav">
        <div className="navbar-nav">
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/marvel"
          >
            Marvel
          </NavLink>

          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/dc"
          >
            DC
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
        <ul className="navbar-nav ml-auto">
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/login"
          >
            Logout
          </NavLink>
        </ul>
      </div>
    </nav>
  );
};
