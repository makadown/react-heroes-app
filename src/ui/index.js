export { Navbar } from './components/NavBar';
