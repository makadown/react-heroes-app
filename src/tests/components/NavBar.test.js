import React from 'react';
import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { NavBar } from '../../components/ui/NavBar';
import { MemoryRouter, Router } from 'react-router';
import { Types } from '../../types/types';

describe('Pruebas de <NavBar/>', () => {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    };

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'test'
        }
    };
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <NavBar />
                </Router>
            </MemoryRouter>

        </AuthContext.Provider>
    );

    afterEach(()=>{
        jest.clearAllMocks();
    });

    test('Debe de mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toContain('test');
    });

    test('debe de llama el logout y usar history', () => {

        wrapper.find('button').at(1).simulate('click');
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: Types.logout
        });

        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    });


});
