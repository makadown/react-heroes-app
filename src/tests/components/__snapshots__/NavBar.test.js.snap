// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <NavBar/> Debe de mostrarse correctamente  1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Brands
  </a>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info mr-5"
      >
        Bienvenid@ 
        test
      </span>
      <button
        className="btn nav-item nav-link"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
